src/fabric/gamma/SplitViewScreenNativeComponent.ts

```diff
@@ -1,17 +1,35 @@
 'use client';
 
 import type { ViewProps } from 'react-native';
-import { WithDefault } from 'react-native/Libraries/Types/CodegenTypes';
+import {
+  DirectEventHandler,
+  WithDefault,
+} from 'react-native/Libraries/Types/CodegenTypes';
 import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';
 
+type GenericEmptyEvent = Readonly<{}>;
+
 type SplitViewScreenColumnType = 'column' | 'inspector';
 
 interface InternalNativeProps extends ViewProps {
   // Config
   columnType?: WithDefault<SplitViewScreenColumnType, 'column'>;
+
+  // Events
+  onWillAppear?: DirectEventHandler<GenericEmptyEvent>;
+  onDidAppear?: DirectEventHandler<GenericEmptyEvent>;
+  onWillDisappear?: DirectEventHandler<GenericEmptyEvent>;
+  onDidDisappear?: DirectEventHandler<GenericEmptyEvent>;
 }
 
-export interface NativeProps extends ViewProps {}
+export interface NativeProps extends ViewProps {
```

Comment:
Why do we have these duplicated? NativeProps & internal native props?

Reply:
I don't want to expose `columnType` to external component

Reply:
outdated by the changes from parent PR